src/importlinter/domain/fields.py

```diff
@@ -109,16 +109,20 @@ def parse(self, raw_data: Union[str, List]) -> Module:
         return Module(StringField().parse(raw_data))
 
 
-class DirectImportField(Field):
+class ImportExpressionField(Field):
     """
-    A field for DirectImports.
+    A field for ImportExpressions.
 
-    Expects raw data in the form: "mypackage.foo.importer -> mypackage.bar.imported".
+    Expects raw data in the form:
+        "mypackage.foo.importer -> mypackage.bar.imported".
+
+    In addition, it handles wildcards:
+        "mypackage.*.importer -> mypackage.bar.imp*"
     """
 
-    def parse(self, raw_data: Union[str, List]) -> DirectImport:
+    def parse(self, raw_data: Union[str, List]) -> ImportExpression:
         string = StringField().parse(raw_data)
         importer, _, imported = string.partition(" -> ")
         if not (importer and imported):
             raise ValidationError('Must be in the form "package.importer -> package.imported".')
-        return DirectImport(importer=Module(importer), imported=Module(imported))
+        return ImportExpression(importer=importer, imported=imported)
```

Comment:
As per [our discussion](https://github.com/seddonym/import-linter/pull/103#discussion_r688307403), I don't think we should support partial wildcards yet - just the ability to use `*` in place of a whole module name. It would be good to raise a ValidationError here if we see something like that.

Reply:
done